generated/serialization.h

```diff
@@ -0,0 +1,80 @@
+#ifndef DTOGEN_SERIALIZATION_H
+#define DTOGEN_SERIALIZATION_H
+
+#include <memory>
+#include <optional>
+
+#include "nlohmann/json.hpp"
+
+// TODO: move elsewhere
+#ifdef __clang__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif __GNUC__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif _MSC_VER
+#define UNREACHABLE_CODE __assume(0);
+#endif
+
+// TODO: move elsewhere
+#include <stdexcept>
+namespace org {
+namespace openapitools {
+namespace model {
+class invalid_enum_value : public std::runtime_error {
+    public:
+        using std::runtime_error::runtime_error;
+};
+}  // namespace org
+}  // namespace openapitools
+}  // namespace model
+
+
+namespace nlohmann {
+
+template <typename T>
+struct adl_serializer<std::optional<T>> {
+    static void to_json(json& j, const std::optional<T>& opt) {
+        opt == std::nullopt ? j = nullptr : j = *opt;
```

Comment:
uhm... despite fancy, this looks fairly dangerous. The result of this expression is whatever is returned from the assignment operator of `nlohmann::json`. I'd feel better if that would be an old-school `if {} else {}`.

... or, if you want to keep some of the fanciness but rely on the STL:

```C++
j = opt.value_or(nullptr);
```